[DX-993, TT-11639] Updated multiple middleware for OAS (#4012)
TykTechnologies/tyk-docs: tyk-docs/content/product-stack/tyk-gateway/middleware/request-header-tyk-oas.md

---
title: Using the Request Header Transform with Tyk OAS APIs
date: 2024-01-20
description: "Using the Request Header Transform middleware with Tyk OAS APIs"
tags: ["Request Header Transform", "middleware", "per-endpoint", "Tyk OAS"]
---

> ## Overview

Author: ```suggestion
```

                    }
                }
            }
        }
    }
}
```

In this example the Request Header Transform middleware has been configured forrequests to the `GET /status/200` endpoint. Any request received to that endpoint will have the `X-Static` header removed and the `X-Secret` header added, with the value set to `the-secret-key-is-secret`.

The configuration above is a complete and valid Tyk OAS API Definition that you can import into Tyk to try out the endpoint-level request header transform.

### Combining API-level and Endpoint-level transforms

If the API-level transform in the previous [example]({{< ref "product-stack/tyk-gateway/middleware/request-header-tyk-oas#api-level-transform" >}}) is applied to the same API, then because the API-level transformation is performed first, the `X-Static` header will be added (by the API-level transform) and then removed (by the endpoint-level transform) such that the overall effect of the two transforms for a call to `GET /status/200` would be to add three headers:
 - `X-Request-ID`
 - `X-User-ID`
 - `X-Secret`

and to remove one:
 - `Auth_Id` 

## Configuring the Request Header Transform in the API Designer

> Adding and configuring the transforms to your API endpoints is easy when using the API Designer in the Tyk Dashboard, simply follow the following steps:

Reviewer: ```suggestion
Adding and configuring the transforms to your API endpoints is easy when using the API Designer in the Tyk Dashboard, simply follow these steps:
```

}
```

> In this example the Request Header Transform middleware has been configured forrequests to the `GET /status/200` endpoint. Any request received to that endpoint will have the `X-Static` header removed and the `X-Secret` header added, with the value set to `the-secret-key-is-secret`.

Reviewer 2: ```suggestion
In this example the Request Header Transform middleware has been configured for requests to the `GET /status/200` endpoint. Any request received to that endpoint will have the `X-Static` header removed and the `X-Secret` header added, with the value set to `the-secret-key-is-secret`.
```

 - `Auth_Id` 

## Configuring the Request Header Transform in the API Designer

> Adding and configuring the transforms to your API endpoints is easy when using the API Designer in the Tyk Dashboard, simply follow the following steps:

Reviewer 2: ```suggestion
Adding and configuring the transforms to your API endpoints is easy when using the API Designer in the Tyk Dashboard:
```
